test/runner.sh

```diff
@@ -28,7 +28,10 @@ mkdir -p dump
 # Hack to grant TEST_PGUSER superuser status so that we can
 # consistently run tests using the same user rather than the
 # current/local user
-${PSQL} $@ -v ECHO=none -c "ALTER USER ${TEST_PGUSER} WITH SUPERUSER;"
+${PSQL} $@ -U `whoami` -v ECHO=none -c "ALTER USER super_user WITH SUPERUSER;"
+${PSQL} $@ -U super_user -v ECHO=none -c "DROP DATABASE single;" >/dev/null 2>&1 || :
```

Comment:
`pg_regress` creates a database for you so I think we should probably use that DB by specifying `--dbname`. I see the need for dropping and recreating to clear things, but would be good if we used the same DB as created by `pg_regress` for clarity and consistency. Also, I think the DB name should passed from the Makefile as a shell variable rather than being hardcoded. We should probably also set `psql` env variables with DB name and test users so, similar to what we do for `DISABLE_OPTIMIZATIONS` and `TEST_TALBESPACE1_PATH`. We should use descriptive names, e.g., `SUPERUSER`, `NOPERMUSER`, `ADMINUSER`, etc. (just giving some examples) and then we can pre-create some users with different permissions to use in tests.